### PR TITLE
Added test for recieving workflow from client. Added test for databas…

### PR DESCRIPTION
**Pull request by clundin55**
…e API

### CODE REVIEW

File: yw_website/yw_rest_services/tests.py [hunk 1]
```diff
@@ -14,3 +16,24 @@ def test_yw_ping(self):
         bad_route = '/save/pig/'
         bad_response = self.client.get(path=bad_route)
         self.assertNotEqual(bad_response.status_code, 200, msg="Invalid route failed")
+
+    def test_save_upload(self):
+        self.user = User.objects.create_user(username='testuser', password='12345')
```
Comment (aniehuser) [inline yw_website/yw_rest_services/tests.py]: do these tests work if run more than once? usernames are supposed to be unique in our database, so you should generate a unique username each time. Look at [this link](https://docs.python.org/3/library/uuid.html) for reference on how to do this

File: yw_website/yw_db/tests.py [hunk 2]
```diff
@@ -1,3 +1,15 @@
 from django.test import TestCase
+from django.test.client import Client
+from yw_db.models import Workflow
 
 # Create your tests here.
+
+class DBTestCase(TestCase):
+    def test_workflow_save(self):
+        test = Workflow.objects.create(pk=1)
```
Comment (aniehuser) [inline yw_website/yw_db/tests.py]: again, does this test work if run more than once? would it work if there were other tests that created a workflow of id 1? You should create the workflow object to ensure that it is unique and does not collide with other database values.

File: yw_website/yw_db/tests.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
 from django.test import TestCase
+from django.test.client import Client
+from yw_db.models import Workflow
```
Comment (aniehuser) [inline yw_website/yw_db/tests.py:3]: Are we using a test database for this or do these tests get run using the actual database?

**Review by aniehuser — CHANGES_REQUESTED** [review]
look at above questions

**Comment by clundin55** [comment]
@aniehuser django creates and tears down a temporary test database when the test command is run.

File: yw_website/yw_rest_services/tests.py (hunk 1 above)
Comment (aniehuser) [reply yw_website/yw_rest_services/tests.py]: Even if a new test database is created every every time test is run, if there are other tests that are created later where somebody else also creates a user with username `testuser`, we could get poor or inconsistent test results

**Comment by aniehuser** [comment]
> django creates and tears down a temporary test database when the test command is run.

@clundin55  thank god that makes our lives a lot easier

File: yw_website/yw_db/tests.py (hunk 2 above)
Comment (aniehuser) [reply yw_website/yw_db/tests.py]: since workflow id auto-increments I still think it is a bad idea to hardcode in an id.

**Comment by clundin55** [comment]
@aniehuser fixed potential collisions

**Comment by aniehuser** [comment]
@clundin55 you still need to implement a uuid for the username mentioned in [this review](https://github.com/aniehuser/senior-design-group10/pull/32#pullrequestreview-182485606). Other than that, this LGTM!

**Comment by clundin55** [comment]
@aniehuser added uuid username

File: yw_website/yw_rest_services/tests.py [hunk 4]
```diff
@@ -18,11 +19,12 @@ def test_yw_ping(self):
         self.assertNotEqual(bad_response.status_code, 200, msg="Invalid route failed")
 
     def test_save_upload(self):
-        self.user = User.objects.create_user(username='testuser', password='12345')
+        username = str(uuid.uuid1())
```
Comment (aniehuser) [inline yw_website/yw_rest_services/tests.py:22]: lets remember to always use uuid v1 for any other tests we might run.

**Review by aniehuser — APPROVED** [review]

**Pull request merged**